scripts/bash/gitlab_helpers.sh

```diff
@@ -0,0 +1,393 @@
+    gitlab-rails runner "Ci::InstanceVariable.new(key: 'CONTAINER_REGISTRY_HOST', value: 'gcr.io').save"
+}
+
+update_cicd_vars () {
+    local logName=$1
+    # Update instance level variables values accroding to the new GCP project and compute hostname
+    logger $logName "INFO" "Updating instance level CI/CD variables"
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CI_SERVER_HOST').update(value: '$INSTANCE_INTERNAL_HOSTNAME')"
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CI_SERVER_URL').update(value: '$INSTANCE_INTERNAL_URL')"
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CI_API_V4_URL').update(value: '$INSTANCE_INTERNAL_API_V4_URL')"
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CONTAINER_REGISTRY_NAMESPACE').update(value: '$GCP_PROJECT_ID').save"
+    gitlab-rails runner "Ci::InstanceVariable.where(key: 'CONTAINER_REGISTRY_HOST').update(value: 'gcr.io').save"
+}
+
+
+seed_instance_reg_token () {
+    local logName=$1
+    # Seed shared runners registartion token
+    GITLAB_RUNNER_REG_SECRET=`curl -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/attributes/gitlab-ci-runner-registration-token-secret`
+    logger $logName "INFO" "Reading shared runners registration token from secret $GITLAB_RUNNER_REG_SECRET"    
+    GITLAB_RUNNER_REG=`gcloud secrets versions access latest --secret=$GITLAB_RUNNER_REG_SECRET`
+    logger $logName "INFO" "Seeding shared runners registration token"
+    errMsg=$(gitlab-rails runner "appset = Gitlab::CurrentSettings.current_application_settings; appset.set_runners_registration_token('$GITLAB_RUNNER_REG'); appset.save!" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg" 
+}
+
+
+seed_gitlab_root_pwd () {
+    local logName=$1
+    # Seed gitlab root password
+    GITLAB_INITIAL_ROOT_PASSWORD_SECRET=`curl -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/attributes/gitlab-initial-root-pwd-secret`
+    logger $logName "INFO" "Reading gitlab root password from secret $GITLAB_INITIAL_ROOT_PASSWORD_SECRET"
+    GITLAB_INITIAL_ROOT_PASSWORD=`gcloud secrets versions access latest --secret=$GITLAB_INITIAL_ROOT_PASSWORD_SECRET`
+    logger $logName "INFO" "Seeding gitlab root password"
+    errMsg=$(gitlab-rails runner "user = User.find_by_username('root'); user.password = '$GITLAB_INITIAL_ROOT_PASSWORD'; user.password_confirmation = '$GITLAB_INITIAL_ROOT_PASSWORD'; user.save!" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
+}
+
+
+
+create_groups () {
+    local logName=$1
+    # Create repo groups (ci, community, private)
+    logger $logName "INFO" "Creating groups: ci, community, private"
+    errMsg=$(gitlab-rails runner "Groups::CreateService.new(User.find_by_id(1), params = {name: 'CI CD Tools', path: 'ci', visibility_level: 10}).execute" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
```

Comment:
I see what you tried to accomplish here, let's try to implement try catch.
If there there are commands that you can continue, just continue without checking error, but do display them (WARNING, DEBUG, ERROR)
If there are commands that are crucial and you need to exit let's do something like this:
{
command1 &&
command2 &&
command3
} || {
catch # Do what need to be done
}

Do you need to do cleanup if something go wrong during deployment phase?

Reply:
See updated commit
https://github.com/SygniaLabs/ScallOps/pull/22/commits/c356495acff23ef9f2a55a71c801aa9af4cbba14